packages/react/src/components/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,290 @@
+
+  static slotClassNames: TooltipSlotClassNames = {
+    content: `${Tooltip.className}__content`,
+  }
+
+  static Content = TooltipContent
+
+  static propTypes = {
+    ...commonPropTypes.createCommon({
+      animated: false,
+      as: false,
+      content: false,
+    }),
+    align: PropTypes.oneOf(ALIGNMENTS),
+    defaultOpen: PropTypes.bool,
+    inline: PropTypes.bool,
+    mountNode: customPropTypes.domNode,
+    mouseLeaveDelay: PropTypes.number,
+    open: PropTypes.bool,
+    onOpenChange: PropTypes.func,
+    pointing: PropTypes.bool,
+    position: PropTypes.oneOf(POSITIONS),
+    renderContent: PropTypes.func,
+    target: PropTypes.any,
+    trigger: customPropTypes.every([customPropTypes.disallow(['children']), PropTypes.any]),
+    content: customPropTypes.shorthandAllowingChildren,
+  }
+
+  static defaultProps: TooltipProps = {
+    align: 'center',
+    mountNode: isBrowser() ? document.body : null,
+    position: 'above',
+    mouseLeaveDelay: 500,
+    pointing: true,
+  }
+
+  static autoControlledProps = ['open']
+
+  pointerTargetRef = React.createRef<HTMLElement>()
+  triggerRef = React.createRef<HTMLElement>() as React.MutableRefObject<HTMLElement>
+  tooltipDomElement = null
+
+  closeTimeoutId
+
+  renderComponent({
+    classes,
+    rtl,
+    accessibility,
+  }: RenderResultConfig<TooltipProps>): React.ReactNode {
+    const { mountNode } = this.props
+    const tooltipContent = this.renderTooltipContent(classes.content, rtl, accessibility)
+
+    return (
+      <>
+        {this.renderTrigger(accessibility)}
+        {mountNode && ReactDOM.createPortal(tooltipContent, mountNode)}
+      </>
+    )
+  }
+
+  getTriggerProps(triggerElement) {
+    const triggerProps: any = {}
+
+    triggerProps.onFocus = (e, ...args) => {
+      if (isFromKeyboard()) {
+        this.trySetOpen(true, e)
+      }
+      _.invoke(triggerElement, 'props.onFocus', e, ...args)
+    }
+    triggerProps.onBlur = (e, ...args) => {
+      if (this.shouldBlurClose(e)) {
+        this.trySetOpen(false, e)
+      }
+      _.invoke(triggerElement, 'props.onBlur', e, ...args)
+    }
+
+    triggerProps.onMouseEnter = (e, ...args) => {
+      this.setTooltipOpen(true, e)
+      setWhatInputSource('mouse')
+      _.invoke(triggerElement, 'props.onMouseEnter', e, ...args)
+    }
+    triggerProps.onMouseLeave = (e, ...args) => {
+      this.setTooltipOpen(false, e)
+      _.invoke(triggerElement, 'props.onMouseLeave', e, ...args)
+    }
+
+    return triggerProps
+  }
+
+  getContentProps = (predefinedProps?) => {
+    const contentHandlerProps: any = {}
+
+    contentHandlerProps.onMouseEnter = (e, contentProps) => {
+      this.setTooltipOpen(true, e)
+      predefinedProps && _.invoke(predefinedProps, 'onMouseEnter', e, contentProps)
+    }
+    contentHandlerProps.onMouseLeave = (e, contentProps) => {
+      this.setTooltipOpen(false, e)
+      predefinedProps && _.invoke(predefinedProps, 'onMouseLeave', e, contentProps)
```

Comment:
```suggestion
      _.invoke(predefinedProps, 'onMouseLeave', e, contentProps)
```

Do we really need condition?